promql/quantile.go

```diff
@@ -357,22 +359,41 @@ func coalesceBuckets(buckets buckets) buckets {
 // bucket with the φ-quantile count, so breaking the monotonicity
 // guarantee causes bucketQuantile() to return undefined (nonsense) results.
 //
-// As a somewhat hacky solution, we calculate the "envelope" of the histogram
-// buckets, essentially removing any decreases in the count between successive
-// buckets. We return a bool to indicate if this monotonicity was forced or not.
-func ensureMonotonic(buckets buckets) bool {
-	forced := false
-	max := buckets[0].count
+// As a somewhat hacky solution, we first silently correct any numerically
+// insignificant (below the requested tolerance and likely to be from floating
+// point precision errors) differences between successive buckets regardless of
+// the direction. Then we calculate the "envelope" of the histogram buckets,
+// essentially removing any decreases in the count between successive buckets.
+// We return a bool to indicate if this monotonicity was forced or not, and
+// another bool to indicate if the precision errors were fixed or not.
+func ensureMonotonicAndCorrectPrecisionErrors(buckets buckets, tolerance float64) (bool, bool) {
```

Comment:
As above, I would change the naming from "correct precision errors" to "ignore small deltas"